hardhat.config.ts

```diff
@@ -37,6 +37,7 @@ const config: HardhatUserConfig = {
     networks: {
         hardhat: {
             gas: 12000000,
+            hardfork: "berlin",
```

Comment:
solidity-coverage now uses [hardhat release v2.6.0](https://github.com/nomiclabs/hardhat/releases/tag/hardhat-core-v2.6.0) by default which uses the London hardfork and enables eip-1559. This was causing the original tests to fail. The workaround was to explicitly set the hardfork version to berlin to fall back to the old behaviour. However, I think the test suite/config should be updated going forward. what do you think? @kyriediculous @yondonfu 

![Screenshot 2021-08-10 at 12 54 22 AM](https://user-images.githubusercontent.com/23727056/128762561-9ffc0319-f592-4f9d-a250-3e85b3b7c9a9.png)

Reply:
> This was causing the original tests to fail. The workaround was to explicitly set the hardfork version to berlin to fall back to the old behaviour. However, I think the test suite/config should be updated going forward. what do you think?

I think pinning the hardfork value to `berlin` in this PR is ok, but agreed that separately we should a) switch the hardfork value to `london` i.e. the default and b) make sure the tests pass when the hardfork value is set to `london`.

@kautukkundan Can you create an issue tracking the above and take a look at that?

> it also drops support for node v10 which is used in github actions in PR

Let's make note of the required Node version in the README.

Regarding #439 - I think we can do the following:

- Keep the Node version used in that PR since it is being merged into `streamflow`
- Rebase `next` on top of `streamflow` and modify the GH action to use a newer Node version supported by hardhat [1]

[1] Or cherry pick the commit. Whichever is easier.

Reply:
> Let's make note of the required Node version in the README.

Bump on the above. After this should be ready to merge!